File: dataset/codenet/Java/Code/s325080396.java
import java.util.Scanner;
class Main {
	static final int MAX = 200000;
	public static void main(String[] args) {
		int[] R = new int[MAX];
		Scanner in = new Scanner(System.in);
		int n = in.nextInt();
		for(int i = 0;i<n;i++) R[i] = in.nextInt();		
		int maxv = R[1]-R[0];
		int minv = R[0];
		for(int i = 1;i<n;i++){
			maxv = Math.max(maxv, R[i]-minv);	
			minv = Math.min(minv, R[i]);
		}
		System.out.println(maxv);
	}
}

File: output/deepseek_chat/codenet/direct_translation/java/python/s325080396.py
MAX = 200000
R = [0] * MAX
n = int(input())
for i in range(n):
    R[i] = int(input())
maxv = R[1] - R[0]
minv = R[0]
for i in range(1, n):
    maxv = max(maxv, R[i] - minv)
    minv = min(minv, R[i])
print(maxv)
